### DISCUSSION TITLE
Build systems, Meson, and friends

### DISCUSSION THREAD

**Discussion by christianparpart** [libsixel/libsixel#60] [General]
Hi guys,

i'm saying it straight, I was almost to create a PR to change the build system to CMake, assuming that it was still using autotools.
But then I realized you switched from autotools to Meson.

I don't want to have any religious war here, not at all. I'm just wondering if there is any reasoning beyond subjective personal preference. When it comes to adaptability I'd always say that CMake is best, because it's most widely supported, including and even on Windows - a platform I for example surprisingly care about. Regardless my personal preferences, I think that for adoptability, using something well known is always better than something that may or may not be the bleeding-edge and state-of-the-art - spweaking from a maintainers and user's point of view. :)

I don't want to force any change, but I'm still curious.

Happy weekend hacks,
Christian Parpart.

**Comment by christianparpart** [comment]
p.s.: Since libsixel is lagging some package system support on Fedora and Windows (vcpkg), I considered embedding libsixel into Contour. That doesn't work when the build system is Meson, unless my google search results are playing with me. :)

**Reply by eli-schwartz** [reply]
It would work quite well if contour used Meson!
And if libsixel used CMake, then other projects which use Meson would have a harder time embedding it into their projects.

All this really says is that if you use the same build system for *everything*, you get vertical integration.

However, it is fully possible to use generic build systems via cmake fetchcontent/externalproject, you do however have to manually specify the configure/build/install commands.
